src/test/ui/pattern/usefulness/empty-match.exhaustive_patterns.stderr

```diff
@@ -1,257 +1,257 @@
 error: unreachable pattern
-  --> $DIR/empty-match.rs:53:9
+  --> $DIR/empty-match.rs:37:9
    |
 LL |         _ => {},
    |         ^
    |
 note: the lint level is defined here
-  --> $DIR/empty-match.rs:6:9
+  --> $DIR/empty-match.rs:8:9
    |
 LL | #![deny(unreachable_patterns)]
    |         ^^^^^^^^^^^^^^^^^^^^
 
 error: unreachable pattern
-  --> $DIR/empty-match.rs:56:9
+  --> $DIR/empty-match.rs:40:9
    |
 LL |         _ if false => {},
    |         ^
 
 error: unreachable pattern
-  --> $DIR/empty-match.rs:63:9
+  --> $DIR/empty-match.rs:47:9
    |
 LL |         _ => {},
    |         ^
 
 error: unreachable pattern
-  --> $DIR/empty-match.rs:66:9
+  --> $DIR/empty-match.rs:50:9
    |
 LL |         _ if false => {},
    |         ^
 
 error: unreachable pattern
-  --> $DIR/empty-match.rs:73:9
+  --> $DIR/empty-match.rs:57:9
    |
 LL |         _ => {},
    |         ^
 
 error: unreachable pattern
-  --> $DIR/empty-match.rs:76:9
+  --> $DIR/empty-match.rs:60:9
    |
 LL |         _ if false => {},
    |         ^
 
-error: unreachable pattern
-  --> $DIR/empty-match.rs:83:9
+error[E0004]: non-exhaustive patterns: type `u8` is non-empty
+  --> $DIR/empty-match.rs:78:20
    |
-LL |         Some(_) => {}
-   |         ^^^^^^^
-
-error: unreachable pattern
-  --> $DIR/empty-match.rs:87:9
+LL |     match_no_arms!(0u8);
+   |                    ^^^
    |
-LL |         Some(_) => {}
-   |         ^^^^^^^
+   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
+   = note: the matched value is of type `u8`
 
-error[E0004]: non-exhaustive patterns: type `u8` is non-empty
-  --> $DIR/empty-match.rs:90:18
+error[E0004]: non-exhaustive patterns: type `NonEmptyStruct1` is non-empty
+  --> $DIR/empty-match.rs:79:20
    |
-LL |     match_empty!(0u8);
-   |                  ^^^
+LL | struct NonEmptyStruct1;
+   | ----------------------- `NonEmptyStruct1` defined here
+...
+LL |     match_no_arms!(NonEmptyStruct1);
+   |                    ^^^^^^^^^^^^^^^
    |
    = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
-   = note: the matched value is of type `u8`
+   = note: the matched value is of type `NonEmptyStruct1`
```

Comment:
I would like us to handle _this case in particular_ (matching on an empty struct) explicitly in the diagnostic output as it might not be obvious to a newcomer.